### PR TITLE
Convert analyzer template test to node:test

### PR DESCRIPTION
**Pull request by justinfagnani**
node:test is pretty nice, and Uvu has some rough edges. This is just a first test: new tests would be written to node:test and existing tests would be ported over, starting with the analyzer.

### CODE REVIEW

File: packages/labs/analyzer/src/test/server/lit-html/template_test.ts [hunk 1]
```diff
@@ -4,53 +4,48 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import {suite} from 'uvu';
+import {test, describe as suite} from 'node:test';
 // eslint-disable-next-line import/extensions
 import * as assert from 'uvu/assert';
```
Comment (augustjk) [inline packages/labs/analyzer/src/test/server/lit-html/template_test.ts:9]: do we also want to replace this with `chai@5` which supports esm?

Comment (justinfagnani) [reply packages/labs/analyzer/src/test/server/lit-html/template_test.ts:9]: oh, node has an assert library too, I just forgot to use it!

**Review by augustjk — APPROVED** [review]

File: packages/labs/analyzer/src/test/server/utils.ts [hunk 2]
```diff
@@ -244,6 +244,30 @@ export const setupAnalyzerForTest = (
   }
 };
 
+export const setupAnalyzerForNodeTest = (lang: Language, pkg: string) => {
```
Comment (rictic) [inline packages/labs/analyzer/src/test/server/utils.ts:247]: Can write setupAnalyzerForTest in terms of setupAnalyzerForNodeTest I think:

```ts
  try {
    Object.assign(ctx, setupAnalyzerForNodeTest(lang, pkg))
  } catch (error) {
    // Uvu has a bug where it silently ignores failures in before and after,
    // see https://github.com/lukeed/uvu/issues/191.
    console.error('uvu before error', error);
    process.exit(1);
  }
}

**Review by rictic — APPROVED** [review]
Nice! One suggestion

File: packages/labs/analyzer/src/test/server/utils.ts (hunk 2 above)
Comment (justinfagnani) [reply packages/labs/analyzer/src/test/server/utils.ts:247]: nice, done!

**Comment by justinfagnani** [comment]
Arg, this is why the new tests aren't running on CI: https://github.com/nodejs/node/issues/50658#issuecomment-1806581766

Should we update Node to 21?

**Pull request merged**